### ISSUE TITLE
About run this code, help

### ISSUE THREAD

**Issue by yiiixuan12** [LMZZML/ST-AE#2]
Hello, I read your paper and now I want to reproduce your code.
 I currently ran gernerate_training_data.py first, and there are three npz files for training set, test set and validation set under data/PEMS08 folder. Then I tried to run the python ae_train.py command, which prompted the runtime error：Expected 2D (unbatched) or 3D (batched) input to conv1d, but got input of size: [64, 32, 170, 13], appears in: filter = self.filter_convs[i](residual) code line

**Comment by LMZZML** [comment]
I cannot determine from your description where exactly the error occurs in your code. If you are referring to line 140 in the file state_model.py, the shape of the variable "residual" should be [batch_size, residual_channels, num_nodes, 13]. Based on the error you mentioned, I suspect this might be due to differences in PyTorch versions. You can try replacing the 1x1 convolutional layers in self.filter_convs with nn.Conv2d, as this would be the correct approach.

**Comment by yiiixuan12** [comment]
I'm sorry to bother you again, I still have some questions：
1、What is your pytorch version?
2, I did not modify this statement: stae_model.py in line 20: self.mlp = torch.nn.Conv2d(c_in, c_out, kernel_size=(1, 1), padding=(0, 0), stride=(1, 1), bias=True).
3, I changed the torch of 1.12.1 and 2.0.1 environment both have the same problem.
4、Lastly, do you execute generate_training_data.py first and then execute ae_train.py in the flow of the code?

**Comment by LMZZML** [comment]
1. My pytorch version is 1.10.0.
2. The statement of line 140 in stae_model.py is correspond to line 93, not line 20. Sorry for the mistake in last comment that it's not a 1x1 convolutional layer. Try to replace the nn.Conv1d to nn.Conv2d from line 93 to 102.
3. Pytorch version 1.10.0+ is enough.
4. Of course. The script ae_train.py can not be runned without the three .npz files.
